src/vs/editor/contrib/documentSymbols/outlineTree.ts

```diff
@@ -278,27 +280,26 @@ export class OutlineFilter implements ITreeFilter<OutlineItem> {
 		[SymbolKind.TypeParameter]: 'showTypeParameters',
 	});
 
-	private readonly _filteredTypes = new Set<SymbolKind>();
-
 	constructor(
 		private readonly _prefix: string,
-		@IConfigurationService private readonly _configService: IConfigurationService,
-	) {
-		this.update();
-	}
-
-	update() {
-		this._filteredTypes.clear();
-		forEach(OutlineFilter.configNameToKind, entry => {
-			const key = `${this._prefix}.${entry.key}`;
-			if (this._configService.getValue<boolean>(key) === false) {
-				this._filteredTypes.add(entry.value);
-			}
-		});
-	}
+		@ITextResourceConfigurationService private readonly _textResourceConfigService: ITextResourceConfigurationService,
+	) { }
 
 	filter(element: OutlineItem): boolean {
-		return !(element instanceof OutlineElement) || !this._filteredTypes.has(element.symbol.kind);
+		const outline = OutlineModel.get(element);
+		let uri: URI | undefined;
+
+		if (outline) {
+			uri = outline.textModel.uri;
+		}
+
+		if (!(element instanceof OutlineElement)) {
+			return true;
+		}
+
+		const configName = (OutlineFilter.kindToConfigName as any)[element.symbol.kind] as string;
```

Comment:
why this casted to ` any`?

Reply:
I honestly don't remember, apologies. I took out the typecasts and got no type errors. Sorry about that! e88573e26b71e1471dd1abc6458c6cb7fefba17d